packages/terra-tag/src/RollupTag.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useRef } from 'react';
+import { injectIntl } from 'react-intl';
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import { v4 as uuidv4 } from 'uuid';
+import classNamesBind from 'classnames/bind';
+import PropTypes from 'prop-types';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import styles from './Tag.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * Determines if the Tag List is rolled up or not.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Callback function triggered on click/key press of the roll-up tag
+   */
+  onSelectRollUp: PropTypes.func,
+  /**
+   * Number of tags that are rolled up.
+   */
+  rollupCount: PropTypes.number,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const RollUpTag = (props) => {
+  const {
+    isCollapsed, onSelectRollUp, intl, rollupCount,
+  } = props;
+  const rollUpTagRef = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnSelectRollUp = (event) => {
+    rollUpTagRef.current.setAttribute('data-terra-rollup-tag-show-focus-styles', 'true');
+    event.preventDefault();
+    event.stopPropagation();
+    onSelectRollUp(event);
+  };
+
+  const handleRollUpTagKeyDown = (event) => {
+    rollUpTagRef.current.setAttribute('data-terra-rollup-tag-show-focus-styles', 'true');
```

Comment:
In a React component, we should not be dependent upon data attributes.  You are interacting directly with the DOM instead of taking advantage of the virtual DOM.

Reply:
Since this TagList is replica of Terra-pills we have used most the same logic here for setting the focus of selected tag item.
Thanks

Reply:
We should not replicate bad practices in a new component.  This is not the proper way to handle this in React.